src/usr/local/www/system_certmanager.php

```diff
@@ -1459,6 +1530,32 @@ function list_csrs() {
 //<![CDATA[
 events.push(function() {
 
+	$('.import_type_toggle').click(function() {
+		var x509 = (this.value === 'x509');
+		hideInput('cert', !x509);
+		setRequired('cert', x509);
+		hideInput('key', !x509);
+		setRequired('key', x509);
+		hideInput('pkcs12_cert', x509);
+		setRequired('pkcs12_cert', !x509);
+		hideInput('pkcs12_pass', x509);
+		hideCheckbox('pkcs12_intermediate', x509);
+	});
+	if ($('input[name=import_type]:checked').val() == 'x509') {
+		hideInput('pkcs12_cert', true);
```

Comment:
This part should also unhide the `cert` and `key` fields, otherwise they won't be visible if you switch to PKCS#12 and back again.

Reply:
fixed